app/models/Enumerable.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json._
+
+trait Enumerable[A] {
+
+  def withName(str: String): Option[A]
+}
+
+object Enumerable {
+
+  def apply[A](entries: (String, A)*): Enumerable[A] =
```

Comment:
This method can be changed to 
`  def apply[A](entries: (String, A)*): Enumerable[A] =
    (str: String) => entries.toMap.get(str)
`

Reply:
but less clear.  Removing the return type, the suggestion fails to compile, whereas the original implementation compiles and runs as intended